### PR TITLE
[Feature] PHATE algorithm 

### PR DESCRIPTION
**Pull request by Danqi7**
## Description
adding an affinity based on the potential distance used in PHATE.
adding the alpha-decay kernel. 
Will do in future PRs:
- [ ] automatic choice of diffusion time
- [ ] powering the diffusion matrix in an efficient way.

## Checklist

- [x] I have read the [**How to Contribute**](https://torchdr.github.io/torchdr.contributing.html) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [ ] All tests passed, and additional code has been **covered with new tests**.
- [ ] I have added the PR to the [**RELEASES.rst**](RELEASES.rst) file.

### CODE REVIEW

**Comment by Danqi7** [comment]
Hi @huguesva , our PR passed all tests except https://github.com/TorchDR/TorchDR/blob/main/torchdr/tests/test_spectral.py#L41 with Error:
`test_spectral.py::test_KernelPCA_sklearn[None] - AssertionError`.

I can also reproduce the same error with the latest TorchDR repo.  The error is `(shapes (10, 10), (10, 9) mismatch)`. My sklearn version is 1.5.1.

**Comment by Danqi7** [comment]
Thank you @huguesva! I will add the example in the examples/directory soon!

**Comment by huguesva** [comment]
Still making progress to robustify the tests, I will let you know asap!

**Comment by huguesva** [comment]
@guillaumehu @Danqi7 I think we have an issue with the matrix_power function for keops lazy tensors. Doing matrix products gives an error.

**Comment by huguesva** [comment]
I think it is not possible yet to compute the matrix multiplication of two Lazy tensors. 
I started an issue here : https://github.com/getkeops/keops/issues/424

If that's ok we can restrict PHATE and NegPotentialAffinity to regular torch tensors. What do you think? 
Thanks!

**Comment by huguesva** [comment]
Alright I have cleaned up things and now the tests pass.

One concern : from what I understand PHATE uses "euclidean" and not "sqeuclidean" as embedding space metric (called metric_out). However when I put "euclidean" the gradient becomes NaN quite easily. 
If you have time could you review all the PHATE code and make sure we use the good loss and good hyperparameters for t, K, eps, alpha etc...

Also for clarity I merged together the alpha-decay and MAGIC affinity, by just adding the alpha parameter to MAGIC.
Is it normal that the PotentialAffinity uses MAGIC as base affinity? It seems that several step like symmetrization and normalization are a bit redundant. I thought PHATE used an unnormalized kernel as base affinity in PotentialAffinity. 

Thanks a lot!
